### ISSUE TITLE
Allow to have mutation protection without reasignment one in no-param-reassign rule

### ISSUE THREAD

**Issue by klis87** [eslint/eslint#9859]
**What version are you using?**
4.13.1

**What did you do?**
I need a rule which will prevent me to do this:
```js
const myFunc = x => {
  x.prop = 1;
  return x;
}
```
So I did `no-param-reassign": [2, { "props": true }]`

**What happened?**
However, now, in function:
```js
const myFunc = x => {
  x = x + 1;
  return x;
}
```
Eslint will complain because I try to reasign `x`.

**What did you expect to happen?**
I want to be able to allow reasigning, but prevent mutation unless I allow it in whitelist `ignorePropertyModificationsFor`

In my opinion reasigning is 100% safe as this doesn't mutate function param (I know it affects `arguments` var but it is usually not a problem), contrary to param mutation as adding/replacing/removing props, which causes a side effect and makes a function impure.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
